src/Uno.UI.Tests/BinderTests/Given_Binder.cs

```diff
@@ -848,6 +848,21 @@ public void When_ExplicitUpdateSourceTrigger_UpdateSource()
 			Assert.AreEqual(-42, target.MyProperty);
 		}
 
+		[TestMethod]
+		public void When_ExplicitSetBindingBetweenProperties_IsNotFallBackValue()
+		{
+			var source = new MyBindingSource { IntValue = 42 };
+			var target = new MyControl();
+			var target2 = new MyObjectTest();
+
+			target.SetBinding(MyControl.MyPropertyProperty, new Binding { Source = source, Path = nameof(MyBindingSource.IntValue), Mode = BindingMode.TwoWay, UpdateSourceTrigger = UpdateSourceTrigger.Explicit });
```

Comment:
You change is making explicit changes with regards to weak references; you should ensure that the source is actually garbage collected to validate the scenario.